### PR TITLE
full rank matrices

### PR DESCRIPTION
**Pull request by annacalissano**
Adding the class for full rank matrices and change the backend adding np.linalg.matrix_rank

### CODE REVIEW

File: geomstats/geometry/full_rank_matrices.py
Comment (nguigs) [inline geomstats/geometry/full_rank_matrices.py:53]: A docstring with Latex instructions (backslash) needs to have an r prefix, i.e. open the docstring with `r""" ... """`

File: geomstats/geometry/full_rank_matrices.py
Comment (nguigs) [inline geomstats/geometry/full_rank_matrices.py:102]: can you use `self.rank` directly? Naming a lambda function is not recommended...

File: geomstats/geometry/full_rank_matrices.py
Comment (nguigs) [inline geomstats/geometry/full_rank_matrices.py:1]: let's use capital E for Euclidean in docstrings

File: geomstats/geometry/full_rank_matrices.py
Comment (nguigs) [inline geomstats/geometry/full_rank_matrices.py:10]: What do you mean by Euclidean entries?

File: geomstats/geometry/full_rank_matrices.py
Comment (nguigs) [inline geomstats/geometry/full_rank_matrices.py:75]: It's not the uniform distribution is it?

File: tests/tests_geomstats/test_full_rank_matrices.py
Comment (nguigs) [inline tests/tests_geomstats/test_full_rank_matrices.py:53]: Can you test vectorization as well? Use e.g. `n_samples>1`

**Review by nguigs — COMMENTED** [review]
Great PR to start with! Just a few comments on some docstrings and to pass DeepSource tests

File: geomstats/_backend/autograd/linalg.py
Comment (annacalissano) [reply geomstats/_backend/autograd/linalg.py:17]: Done

File: geomstats/geometry/full_rank_matrices.py
Comment (annacalissano) [reply geomstats/geometry/full_rank_matrices.py:53]: Done

File: geomstats/geometry/full_rank_matrices.py
Comment (annacalissano) [reply geomstats/geometry/full_rank_matrices.py:10]: taking values in R. I change it to Class for R_*(m,n) matrices of dimension m x n and full rank.

File: geomstats/geometry/full_rank_matrices.py
Comment (annacalissano) [reply geomstats/geometry/full_rank_matrices.py:75]: no it is a gaussian distribution

File: tests/tests_geomstats/test_full_rank_matrices.py
Comment (annacalissano) [reply tests/tests_geomstats/test_full_rank_matrices.py:53]: done

File: geomstats/_backend/numpy/linalg.py
Comment (ninamiolane) [inline geomstats/_backend/numpy/linalg.py:14]: This should be sorted in alphabetic order (to avoid importing the same function twice)

File: geomstats/geometry/full_rank_matrices.py
Comment (ninamiolane) [inline geomstats/geometry/full_rank_matrices.py:9]: Nice!!

**Review by ninamiolane — APPROVED** [review]
Very nice, thank you! Just one mini-comment: the imports should be in alphabetic order in the backends. Happy to merge once this is addressed, or we can merge now and you address this in another PR.

Thank you!! 🙌

**Comment by ninamiolane** [comment]
@annacalissano merging now, I'll let you correct the alphabetic order of the imports in the backend in the next PR!

**Pull request merged**
